Assets/Scripts/BuddyManager.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+public class BuddyManager : SingletonBehaviour<BuddyManager> {
+
+	List<BuddyStats> _buddyStats = new List<BuddyStats>();
+
+	public static void RegisterBuddy( BuddyStats buddyStats )
+	{
+		instance._buddyStats.Add( buddyStats );
+	}
+
+	public static void DecrementAllBuddyResources()
+	{
+		foreach ( BuddyStats buddyStat in instance._buddyStats )
+		{
+			buddyStat.DecrementResources();
+		}
+
+		instance._buddyStats.RemoveAll( buddyStat => buddyStat.isAlive == false ); // Predicate statement
```

Comment:
Dang this is cool. I didn't really understand the use of lambdas/predicates in c# until seeing this an looking it up. Thought it only worked w/ LINQ